scheduler/plugin/plugin.go

```diff
@@ -275,13 +275,28 @@ func (pl *wasmPlugin) PostFilter(ctx context.Context, state *framework.CycleStat
 var _ framework.PreScorePlugin = (*wasmPlugin)(nil)
 
 // PreScore implements the same method as documented on framework.PreScorePlugin.
-func (pl *wasmPlugin) PreScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) *framework.Status {
+func (pl *wasmPlugin) PreScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, nodes []*v1.Node) (status *framework.Status) {
 	// We implement PreScorePlugin with ScorePlugin, even when the guest doesn't.
 	if pl.guestInterfaces&iPreScorePlugin == 0 {
 		return nil // unimplemented
 	}
 
-	panic("TODO: scheduling: PreScore")
+	// Add the stack to the go context so that the corresponding host function
+	// can look them up.
+
+	// Use v1.NodeList to encode the nodes, as it is easier for both sides.
```

Comment:
I will make one change to lazy encode this, only when asked by the guest. otherwise this PR is stable